Add support for OpenAPI version 3.0.4

### Is your feature request related to a problem?

Libraries are starting to roll out with this version,  would be nice to support it here. Currently getting

> 08:41:32 [ERROR] [starlight-openapi-plugin] Unsupported OpenAPI version: 3.0.4. Swagger Parser only supports versions 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.1.0
> Unsupported OpenAPI version: 3.0.4. Swagger Parser only supports versions 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.1.0

### Describe the solution you'd like

Upgrade `https://www.npmjs.com/package/@readme/openapi-parser` to latest version

### Describe alternatives you've considered

Download the openapi json schema to filesystem and manually change the version at the top to 3.0.3

### Additional Context

_No response_

--- a/.changeset/ninety-bags-cry.md
+++ b/.changeset/ninety-bags-cry.md
@@ -1,0 +1,5 @@
+---
+'starlight-openapi': minor
+---
+
+Adds support for schemas using version `3.0.4` of the OpenAPI specification by updating the OpenAPI and Swagger API definitions parser used internally.

--- a/packages/starlight-openapi/package.json
+++ b/packages/starlight-openapi/package.json
@@ -17,7 +17,7 @@
     "lint": "eslint . --cache --max-warnings=0"
   },
   "dependencies": {
-    "@readme/openapi-parser": "^2.5.0",
+    "@readme/openapi-parser": "^2.7.0",
     "github-slugger": "^2.0.0",
     "url-template": "^3.1.1"
   },

